propagators/aws/xray/idgenerator.go

```diff
@@ -19,16 +19,34 @@ import (
 
 // IDGenerator is used for generating a new traceID and spanID.
 type IDGenerator struct {
-	sync.Mutex
+	mu         sync.Mutex
 	randSource *rand.Rand
 }
 
 var _ sdktrace.IDGenerator = &IDGenerator{}
 
+// Lock is used to lock the IDGenerator for concurrent use.
+// Deprecated: mutex is internal and shall not be used.
```

Comment:
```suggestion
// Lock is used to lock the IDGenerator for concurrent use.
//
// Deprecated: mutex is internal and shall not be used.
```

Deprecated A line needs to be retained, otherwise the deprecation identifier cannot be recognized by Go documentation and IDE. https://tip.golang.org/doc/comment#deprecations